### DISCUSSION TITLE
A way to invalidate useCan provider

### DISCUSSION THREAD

**Discussion by vinclou** [refinedev/refine#3160] [Q&A] [answered]
### Is there a way to invalidate all useCan queries? From turning from fresh to stale >

`Tried code below with various different options, but it never went from green to yellow, on my handler to change, perms`

```tsx
  invalidate({
    invalidates: ['all'],
    dataProviderName: 'useCan'
  })
```

**Comment by vinclou** [comment, marked as answer]
I actually had a React state sync problem and was able to solve it without invalidating useCan queries; Although it would be interesting to hear if that's even possible =)

**Reply by yildirayunlu** [reply]
Hey @vinclou thanks for your feedback 🙏🏼 

You can invalidate the `refetch` usage if it is in the same scope. You can check [here](https://refine.dev/docs/faq/#how-can-i-refetch-data).

Another option would be to use [`invalidateQueries`](https://tanstack.com/query/v4/docs/guides/query-invalidation) from react-query. However, we do not export the entire react-query yet. If you can open a **feature request**, we'll take care of it. 🚀
